### PR TITLE
[c10d] Expose check method to Python for store via pybind

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116144

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @wz337 @tianyu-l @wconstab @yf225

Differential Revision: [D52310987](https://our.internmc.facebook.com/intern/diff/D52310987)

### CODE REVIEW

File: torch/csrc/distributed/c10d/init.cpp [hunk 1]
```diff
@@ -1033,6 +1033,28 @@ Example::
...
     >>> # Should return 7
     >>> store.get("first_key")
+)")
+          .def(
+              "check",
+              &::c10d::Store::check,
+              py::call_guard<py::gil_scoped_release>(),
+              R"(
+The call to check whether a given list of ``keys`` have value stored in
+the store. This call is nonblocking.
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/init.cpp:1043]: it says its nonblocking.  But did you find a way to make it true? (if server does not respond, do we return early?)

Comment (fduwjj) [reply torch/csrc/distributed/c10d/init.cpp:1043]: Well currently, I don't have a very strong proof to say it is 100% non-blocking. I think we have a plan to fix it. But to unblock the dump work, maybe I should just remove it?

**Comment by fduwjj** [comment]
@fduwjj has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D52310987).

File: torch/csrc/distributed/c10d/init.cpp (hunk 1 above)
Comment (fduwjj) [reply torch/csrc/distributed/c10d/init.cpp:1043]: For example, by looking at HashStore, FileStore and TCPStore's code, aside from the mutex lock, there is no blocking logic, here.. Maybe, we should not say non-blocking for now

**Comment by fduwjj** [comment]
@fduwjj has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D52310987).

**Review by wconstab — APPROVED** [review]

**Pull request closed without merging**